### ISSUE TITLE
unit tests, assert true inside IF

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#874]
```abap
IF something1 AND something2.
  cl_abap_unit_assert=>assert_true( abap_true ).
ELSE.
  cl_abap_unit_assert=>fail( ).
ENDIF.
```

can be rewritten to something like

```abap
cl_abap_unit_assert=>assert_true( something1 ).
cl_abap_unit_assert=>assert_true( something2 ).
```

or is it a matter of finding static predetermined assert method calls? ie is there a use case for `cl_abap_unit_assert=>assert_true( abap_true ).` ? 

I use `cl_abap_unit_assert=>fail( ).` only(?) before CATCH to check execptions are raised
